src/build/stable/commands/compile/wasm_binary/rust/stable_canister_template/src/wasm_binary_manipulation.rs

```diff
@@ -36,24 +38,39 @@ pub fn get_js_code() -> Vec<u8> {
 #[inline(never)]
 #[no_mangle]
 extern "C" fn init_wasm_data_passive_data(wasm_data_vec_location: i32) -> usize {
-    "123_456_789".parse::<usize>().unwrap() + wasm_data_vec_location as usize // TODO must be like this for weird optimization reasons
+    // This is to prevent compiler optimizations that interfere with the Wasm binary manipulation
+    std::env::var("init_wasm_data_passive_data").map_or(0, |s| s.len())
+        + wasm_data_vec_location as usize
 }
 
-// TODO seems we need to do this to stop the compiler from hard-coding the result of this function where it is called
-// TODO hopefully there's a less hacky way to do this
 #[inline(never)]
 #[no_mangle]
 extern "C" fn wasm_data_passive_data_size() -> usize {
-    "123_456_789".parse().unwrap()
+    // This is to prevent compiler optimizations that interfere with the Wasm binary manipulation
+    std::env::var("wasm_data_passive_data_size").map_or(0, |s| s.len())
 }
 
 // TODO waiting on license inspired from https://github.com/adambratschikaye/wasm-inject-data/blob/main/src/static_wasm.rs
-pub fn get_wasm_data() -> WasmData {
+pub fn get_wasm_data() -> Result<WasmData, Box<dyn Error>> {
     let size = wasm_data_passive_data_size();
     let mut wasm_data_vec = vec![243; size];
     let wasm_data_vec_location = wasm_data_vec.as_mut_ptr() as i32;
 
     init_wasm_data_passive_data(wasm_data_vec_location);
 
-    serde_json::from_str(std::str::from_utf8(&wasm_data_vec).unwrap()).unwrap()
+    let wasm_data_str = std::str::from_utf8(&wasm_data_vec).map_err(|e| {
+        format!(
+            "WasmData conversion failed while converting Vec<u8> to String: {}",
+            e
+        )
+    })?;
+
+    let wasm_data: WasmData = serde_json::from_str(wasm_data_str).map_err(|e| {
```

Comment:
qualifying? I'm not sure about this one... I guess I'm also not sure about the std:str ones either...

Reply:
I have decided that in some cases a little qualification is okay if it will help with understanding...like from_str, env::var...imagine if it just said var, it would be a bit confusing...so I think generally we shouldn't qualify, but in one off cases we can decide if it helps readability.